Google.Cloud.Spanner.NHibernate.Tests/NHibernateMockServerTests.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.Connection.MockServer;
+using Google.Cloud.Spanner.NHibernate.Tests.Entities;
+using Google.Cloud.Spanner.V1;
+using Google.Protobuf.WellKnownTypes;
+using NHibernate;
+using Xunit;
+
+namespace Google.Cloud.Spanner.NHibernate.Tests
+{
+    /// <summary>
+    /// Tests CRUD operations using an in-mem Spanner mock server.
+    /// </summary>
+    public class NHibernateMockServerTests : IClassFixture<NHibernateMockServerFixture>
+    {
+        private readonly NHibernateMockServerFixture _fixture;
+
+        public NHibernateMockServerTests(NHibernateMockServerFixture service)
+        {
+            _fixture = service;
+            service.SpannerMock.Reset();
+        }
+
+        [Fact]
+        public async Task VersionNumberIsAutomaticallyGeneratedOnInsertAndUpdate()
+        {
+            var insertSql = "INSERT INTO SingerWithVersion (Version, FirstName, LastName, SingerId) VALUES (@p0, @p1, @p2, @p3)";
+            _fixture.SpannerMock.AddOrUpdateStatementResult(insertSql, StatementResult.CreateUpdateCount(1L));
+
+            var singer = new SingerWithVersion { SingerId = 1L, FirstName = "Pete", LastName = "Allison" };
+
+            using (var session = _fixture.SessionFactory.OpenSession())
+            {
+                using (var transaction = session.BeginTransaction())
+                {
+                    await session.SaveAsync(singer);
+                    await session.FlushAsync();
```

Comment:
nit: This flush call is not really needed. The commit on the following line will automatically also cause a flush. (I know that there are quite a lot of flush calls in many other tests as well, but those are mostly there when there is no transaction)